program/src/state.rs

```diff
@@ -1251,60 +1173,54 @@ mod test_lido {
     fn test_split_reward() {
         let mut spec = RewardDistribution {
             treasury_fee: 3,
-            validation_fee: 2,
             developer_fee: 1,
             st_sol_appreciation: 0,
         };
 
         assert_eq!(
             // In this case the amount can be split exactly,
             // there is no remainder.
-            spec.split_reward(Lamports(600), 1).unwrap(),
+            spec.split_reward(Lamports(600)).unwrap(),
             Fees {
-                treasury_amount: Lamports(300),
-                reward_per_validator: Lamports(200),
-                developer_amount: Lamports(100),
+                treasury_amount: Lamports(450),
+                developer_amount: Lamports(150),
                 st_sol_appreciation_amount: Lamports(0),
             },
         );
 
         assert_eq!(
             // In this case the amount cannot be split exactly, all fees are
             // rounded down.
-            spec.split_reward(Lamports(1_000), 4).unwrap(),
+            spec.split_reward(Lamports(1_000)).unwrap(),
```

Comment:
```suggestion
            spec.split_reward(Lamports(1_003)).unwrap(),
```

The point of this test case was to have an amount that is not a multiple of the denominator, such that the split cannot be exact. In that case all fees should round down, and the remainder goes to stSOL appreciation.

Previously the fee shares summed to 6, and 1000 is not a multiple of 6, so this test exercised the rounding. But now the fee shares sum to 4, and 1000 is a multiple of 4. We can change the amount to split to e.g. 1003. Then we have 3 additional lamports to distribute. 3/4 = 2.25 of those should go to the treasury, which then rounds down to 2. 1/4 = 0.75 of those should go to the developer, which rounds down to 0. But then so far we distributed 2 lamports, and we had 3 to distribute, so the remaining 1 lamport goes to stSOL appreciation.

Reply:
done